servicetalk-examples/http/redirects/src/main/java/io/servicetalk/examples/http/redirects/RedirectWithStateUrlClient.java

```diff
@@ -15,95 +15,37 @@
-import io.servicetalk.http.api.StreamingHttpRequester;
-import io.servicetalk.http.api.StreamingHttpResponse;
 import io.servicetalk.http.netty.HttpClients;
 import io.servicetalk.test.resources.DefaultTestCerts;
 import io.servicetalk.transport.api.ClientSslConfigBuilder;
 
-import static io.servicetalk.concurrent.api.AsyncContextMap.Key.newKey;
 import static io.servicetalk.examples.http.redirects.RedirectingServer.CUSTOM_HEADER;
-import static io.servicetalk.http.api.HttpHeaderNames.CONTENT_LENGTH;
-import static io.servicetalk.http.api.HttpHeaderNames.TRANSFER_ENCODING;
-import static io.servicetalk.http.api.HttpHeaderValues.CHUNKED;
+import static io.servicetalk.http.api.HttpRequestMethod.GET;
 import static io.servicetalk.http.api.HttpRequestMethod.POST;
-import static io.servicetalk.http.api.HttpResponseStatus.StatusClass.REDIRECTION_3XX;
-import static io.servicetalk.http.api.HttpSerializers.textSerializerUtf8;
+import static io.servicetalk.http.api.HttpSerializers.textSerializerAscii;
 
 /**
  * Async `Hello World` example that demonstrates how redirects can be handled automatically by a
  * {@link HttpClients#forMultiAddressUrl() multi-address} client. It demonstrates how users can preserve headers and
  * payload body of the original request while redirecting.
- *
- * For security reasons, headers and payload body are not automatically populated for the redirect request. Users have
- * to explicitly cary the original state when they are sure that redirect does not forward to a malicious target server.
- *
- * For backward compatibility with legacy HTTP clients, ServiceTalk preserves behavior for
- * {@link HttpRequestMethod#POST} requests described in RFC7231, sections
- * <a href="https://datatracker.ietf.org/doc/html/rfc7231#section-6.4.2">6.4.2</a> and
- * <a href="https://datatracker.ietf.org/doc/html/rfc7231#section-6.4.3">6.4.3</a>. For these status codes, manual
- * action may be required to preserve the original request method.
+ * <p>
+ * For security reasons, request methods other than {@link HttpRequestMethod#GET GET} or
+ * {@link HttpRequestMethod#HEAD HEAD}, headers and message body are not automatically redirected for non-relative
+ * locations. Users have to explicitly configure what should be redirected when they are sure that redirect does not
+ * forward to a malicious target server. Relative redirects always carry forward headers and message body.
  */
 public final class RedirectWithStateUrlClient {
 
-    private static final AsyncContextMap.Key<StreamingHttpRequest> ORIGINAL_REQUEST = newKey();
-
     public static void main(String... args) throws Exception {
         try (HttpClient client = HttpClients.forMultiAddressUrl()
-                // This is an optional configuration that applies more restrictive limit for the number or redirects:
-                .maxRedirects(1)
+                // Enables redirection with a custom configuration:
+                .followRedirects(config -> config.maxRedirects(3)
```

Comment:
No, `MultiAddressHttpClientBuilder#maxRedirects(int)` was not the reason, it's deprecated and will go away. Actual reasons:
1. `Consumer` of `RedirectConfiguration` API allows me to pre-configure something (`allowNonRelativeRedirects(true)` in this case) before I give it to users. I followed an approach of `SingleAddressInitializer` instead of `SslConfig` to be consistent at `MultiAddressHttpClientBuilder` API.
2. It also gives me an opportunity to have a single place for javadoc of all these options, instead of duplicating javadoc between `RedirectConfig` and `RedirectConfigBuilder`.
3. Proposed approach requires `RedirectConfigBuilder` to be in `http-api` module. Otherwise, it's weird if `MultiAddressHttpClientBuilder#followRedirects` requires `RedirectConfig` but there is no provided API to build that config. But the filter is in `http-utils`. It complicates the filter if `RedirectConfigBuilder` is too far away from the filter and limits how the redirecting filter can extend options.

Reply:
Followed you suggestion, fixed in this commit: https://github.com/apple/servicetalk/pull/1792/commits/cc2a1116c1abc0827568d2bac6e72fa2e1d7a444